### PR TITLE
[HDF5] Update to 1.14

### LINKED ISSUE #29580: [hdf5] update to 1.14
https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_14_0

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Fixes #29580

### CODE REVIEW

**Comment by dg0yt** [comment]
Can we add a `vcpkg_port_config.cmake` which enables mpi for vtk via appending to `VPCKG_CMAKE_CONFIGURE_OPTIONS`? #22892...

**Comment by dg0yt** [comment]
> > Can we add a vcpkg_port_config.cmake which enables mpi for vtk via appending to VPCKG_CMAKE_CONFIGURE_OPTIONS FEATURES? #22892...
> 
> nope since `FEATURES` also land in `vcpkg_abi_info`

Hm. Isn't the hdf5 dependency also part of the abi info? 
If yes, with the actual feature set or with the requested features?

**Comment by Neumann-A** [comment]
> Hm. Isn't the hdf5 dependency also part of the abi info?

Yes it is with all the requested features. Currently it is not possible for ports in vcpkg to force downstream features cleanly and as such I won't be adding hacks to do it ;)

**Comment by dg0yt** [comment]
Fair enough. No hacks required.

**Comment by dg0yt** [comment]
> > Hm. Isn't the hdf5 dependency also part of the abi info?
> 
> Yes it is with all the requested features. 

What I do see iwhen experimenting is: downstream usage has different hashes for hdf5, depending on the actual set of hdf5 features.

**Comment by Cheney-W** [comment]
Feature `map` and `tools` passed with following triplets:
x86-windows 
x64-windows 
x64-windows-static

**Review by Cheney-W — DISMISSED** [review]

File: ports/vtk/9690.diff [hunk 1]
```diff
@@ -0,0 +1,27 @@
+diff --git a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfH5Driver.cxx b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfH5Driver.cxx
+index 1a412b56d028ecc7015976cd6effe8b0bb59e6fe..8f5e6e4c077b52eb956bfe8aacf7d1647860d405 100644
+--- a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfH5Driver.cxx
++++ b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfH5Driver.cxx
+@@ -156,6 +156,9 @@ static herr_t H5FD_dsm_write(H5FD_t *_file, H5FD_mem_t type, hid_t fapl_id, hadd
```
Comment (JavierMatosD) [inline ports/vtk/9690.diff:5]: @Neumann-A, is there a reason this is .diff format instead of a patch file?

**Pull request merged**